### DISCUSSION TITLE
Target Bootstrap accordion from a different page

### DISCUSSION THREAD

**Discussion by leetempest** [twbs/bootstrap#38010] [Q&A]
I have a page with a Bootstrap Accordion. I would like to link to one of the specific options from another page. So when the visitor clicks this link it will load the accordion page and automatically open the correct option.

So on my link page I have an href link that ends with the tag #collapseGeneralThree. My sample code is shown below:

<div id="accordionGeneral" class="accordion">
<div class="accordion-item" itemscope="itemscope" itemtype="https://schema.org/Question" itemprop="mainEntity">
    <h2 id="headingGeneralOne" class="accordion-header" itemprop="name">
        <button type="button" class="accordion-button collapsed" data-bs-toggle="collapse" data-bs-target="#collapseGeneralOne" aria-expanded="false" aria-general="collapseGeneralOne">GeneralOne Title</button>
    </h2>
    <div id="collapseGeneralOne" class="accordion-collapse collapse" itemscope="itemscope" itemtype="https://schema.org/Answer" itemprop="acceptedAnswer" aria-labelledby="headingGeneralOne" data-bs-parent="#accordionGeneral">
        <div class="accordion-body" itemprop="text">GeneralOne Content.</div>
    </div>
</div>
<div class="accordion-item" itemscope="itemscope" itemtype="https://schema.org/Question" itemprop="mainEntity">
    <h2 id="headingGeneralTwo" class="accordion-header" itemprop="name">
        <button type="button" class="accordion-button collapsed" data-bs-toggle="collapse" data-bs-target="#collapseGeneralTwo" aria-expanded="false" aria-general="collapseGeneralTwo">GeneralTwo Title</button>
    </h2>
    <div id="collapseGeneralTwo" class="accordion-collapse collapse" itemscope="itemscope" itemtype="https://schema.org/Answer" itemprop="acceptedAnswer" aria-labelledby="headingGeneralTwo" data-bs-parent="#accordionGeneral">
        <div class="accordion-body" itemprop="text">GeneralTwo Content.</div>
    </div>
</div>
<div class="accordion-item" itemscope="itemscope" itemtype="https://schema.org/Question" itemprop="mainEntity">
    <h2 id="headingGeneralThree" class="accordion-header" itemprop="name">
        <button type="button" class="accordion-button collapsed" data-bs-toggle="collapse" data-bs-target="#collapseGeneralThree" aria-expanded="false" aria-general="collapseGeneralThree">GeneralThree Title</button>
    </h2>
    <div id="collapseGeneralThree" class="accordion-collapse collapse" itemscope="itemscope" itemtype="https://schema.org/Answer" itemprop="acceptedAnswer" aria-labelledby="headingGeneralThree" data-bs-parent="#accordionGeneral">
        <div class="accordion-body" itemprop="text">GeneralThree Content.</div>
    </div>
</div>

I then have a piece of Javascript running on this page placed just above the tag:

    // Opens accordion automatically if an accordion target is accessed from another page
    // Assumes the accordion-group is the target linked to
    function openAnchorAccordion() {
    if (window.location.hash) {
    var jQuerytarget = jQuery('body').find(window.location.hash);
    //console.log( jQuerytarget );
    if (jQuerytarget.hasClass('collapse')) {
    var jQuerytargetAccordion = jQuerytarget.find('.collapse');
    console.log( jQuerytargetAccordion );
    jQuerytarget.collapse('show');
    }
    }
    }
    openAnchorAccordion();

I can see in the console log if finds the correct tag but cannot figure out why it is not then adding the show class to the releevant div.

Can anyone help?

Thanks

**Comment by nkdas91** [comment]
Hi @leetempest 

Try calling the ```openAnchorAccordion()``` function after the DOM is ready.

```javascript
$(function() {
  openAnchorAccordion();
});
```

Check this comment: https://github.com/twbs/bootstrap/issues/32132#issuecomment-726066766

**Reply by leetempest** [reply]
nkdas91,
Thanks for the rpely, I have treid that but still no luck! :-(

Would using somthing like addClass not work?

**Reply by nkdas91** [reply]
@leetempest, do you get any error in the console?

Does the ```jQuerytarget``` correspond to ```<div class="accordion-collapse collapse"```

See this CodePen demo: https://codepen.io/nkdas91/pen/wvxZyrV?editors=1010
It dynamically opens the third collapse.

**Comment by leetempest** [comment]
nkdas91,
This is my page with the Bootstrap accordion on:

https://kyntronics.under-development.info/knowledge-center/faqs-for-smart-elector-hydraulic-actuator

From this page:

https://kyntronics.under-development.info/assembly

I have a plain text link setup as [testlink] in the left column which is setup as an href to (https://kyntronics.under-development.info/knowledge-center/faqs-for-smart-elector-hydraulic-actuator#collapseGeneralFour)

I don't get any errors in console but do get warnings but I donlt think they are related to this issue.

I appreciate your help.

Lee

**Reply by nkdas91** [reply]
@leetempest, 

Calling ```openAnchorAccordion();``` from the console after the page is loaded does open the collapse.
This means the function is being called before the collapse is ready, no idea why.

Can you try adding a delay? I know this is not a good solution.

```javascript
jQuery(function() {
  setTimeout(openAnchorAccordion(), 2000);
});
```

**Comment by ExtravagantBoost** [comment]
use the .click() function within the code to simulate a click on the respective menus on the accordion, unfortunately, you cannot just add a class to the accordion as it will have no effect, since the accordion cannot be initiated as an AccordionObj (please @mdo 🙏🏻 make this a thing), you cannot just programatically open each menu through the object

**Reply by ExtravagantBoost** [reply]
to also compensate for load time, use an EventListener to check if the page is loaded first, before calling .click()
example:
```javascript
//find accordion element
let myAccordion = document.querySelector(`.accordion`);
//get all menus
myAccordion.setProperty(`menus`,{
    value:myAccordion.querySelectorAll(`.accordion-item`)//all collapsible menus
})
myAccordion.setProperty(`toggle`,{
    value:(n) => {
        if (n != abs(n) /*check whether n is a real number*/) throw Error(`n cannot be a negative number!`)
        myAccordion.menus[n].click()
    }
})
document.addEventListener(`load`,{
    if (window.location.hash = `accordionmenu3`) myAccordion.toggle(3)
})
//wrote all of this without env
```
feel free to edit any of this code to your liking, took me 10 mins to write though :,)
